internal/services/storage/storage_account_customer_managed_key_resource.go

```diff
@@ -49,13 +49,22 @@ func resourceStorageAccountCustomerManagedKey() *pluginsdk.Resource {
 			"key_vault_id": {
 				// TODO: should this be split into two resources, since Key Vault and Managed HSM behave subtly differently in places already
 				Type:     pluginsdk.TypeString,
-				Required: true,
+				Optional: true,
 				ValidateFunc: validation.Any(
 					// Storage Account Customer Managed Keys support both Key Vault and Key Vault Managed HSM keys:
 					// https://learn.microsoft.com/en-us/azure/storage/common/customer-managed-keys-overview
 					commonids.ValidateKeyVaultID,
 					managedhsms.ValidateManagedHSMID,
 				),
+				ExactlyOneOf: []string{"key_vault_id", "key_vault_uri"},
+			},
+
+			"key_vault_uri": {
```

Comment:
FWIW we're intentionally not exposing `key_vault_uri` as a field across the Provider, since we're unable to determine within the Provider when the Key Vault is (temporarily) inaccessible or (permanently) gone - so unfortunately that's not something we can ship without introducing other issues (and even if we made an exception for this resource, we'd get asks for this on other resources, which would have the same issue).

Instead, we should be able to look this up given the `tenant_id`, which would solve this in a different manner - WDYT?

Reply:
Thanks for the explanation @Sewci0. Looking at the [docs](https://learn.microsoft.com/en-us/azure/storage/common/customer-managed-keys-configure-cross-tenant-new-account?tabs=azure-portal#create-a-new-storage-account-encrypted-with-a-key-from-a-different-tenant) for this configuration, it seems like referencing the URI is the best/only way forward in this case, so I think it's ok to add this.